clients/formdata.ts

```diff
@@ -0,0 +1,91 @@
+function escapeDots(key: string): string {
+  return key.replace(/\./g, "\\.");
+}
+
+function unescapeDots(key: string): string {
+  return key.replaceAll(/\\\./g, ".");
+}
+
+export function propsToFormData(props: unknown): FormData {
+  if (props instanceof FormData) {
+    return props;
+  }
+
+  const formData = new FormData();
+
+  const appendToFormData = (
+    _key: string,
+    value: unknown,
```

Comment:
since we don't have conversion type yep, maybe require value as string | object | undefined, and avoid other types.